etna/transforms/decomposition/change_points_based/change_points_models/base.py

```diff
@@ -35,9 +38,13 @@ def get_change_points(self, df: pd.DataFrame, in_column: str) -> List[pd.Timesta
         pass
 
     @staticmethod
-    def _build_intervals(change_points: List[pd.Timestamp]) -> List[TTimestampInterval]:
+    def _build_intervals(change_points: List[Union[pd.Timestamp, int]], dtype: Any) -> List[TTimestampInterval]:
         """Create list of stable intervals from list of change points."""
-        change_points.extend([pd.Timestamp.min, pd.Timestamp.max])
+        if pd.api.types.is_integer_dtype(dtype):
+            change_points.extend([np.iinfo(dtype).min, np.iinfo(dtype).max])
```

Comment:
As I showed in other comment, `.loc` works differently and it mimics the logic with timestamp. The possible problem is if `dtype` will have not too extreme values, like int8, but it shouldn't happen on its own probably, because pandas always uses int64 for integer indices.